docs: updated the navigation url of content on video (#20800)
mdn/content: files/en-us/learn/performance/video/index.md

## Optimizing video delivery

> The sections below describe the following optimization techniques:
> - [compress all video](#compress_all_videos)
> - [optimize `<source>` order](#optimize_source_order)
> - [set autoplay](#video_autoplay)
> - [remove audio from muted video](#remove_audio_from_muted_hero_videos)
> - [optimize video preload](#video_preload)
> - [consider streaming](#consider_streaming)

Reviewer: ```suggestion
The sections below describe the following optimization techniques:

- [compress all video](#compress_all_videos)
- [optimize `<source>` order](#optimize_source_order)
- [set autoplay](#video_autoplay)
- [remove audio from muted video](#remove_audio_from_muted_hero_videos)
- [optimize video preload](#video_preload)
- [consider streaming](#consider_streaming)
```

## Optimizing video delivery

> It's best to [compress all video](#compress), [optimize `<source>` order](#omptimize), set [autoplay](#video_autoplay), [remove audio from muted video](#muted), [optimize video preload](#video_preload), and [consider streaming](#consider_streaming) the video. The sections below describe each of these optimization techniques.

Reviewer: ```suggestion
The sections below describe the following optimization techniques:
- [compress all video](#compress_all_videos)
- [optimize `<source>` order](#optimize_source_order)
- [set autoplay](#video_autoplay)
- [remove audio from muted video](#remove_audio_from_muted_hero_videos)
- [optimize video preload](#video_preload)
- [consider streaming](#consider_streaming)
```

Some of these links were still broken, and I think a list is going to be easier to read.
